src/Donations/resources/DonationDetails/app/components/PaymentInformation/Fields/TimePicker/index.tsx

```diff
@@ -0,0 +1,107 @@
+import {useState} from 'react';
+import {__} from '@wordpress/i18n';
+
+import {useFormContext, useWatch} from 'react-hook-form';
+import {format, parse} from 'date-fns';
+
+import Field from '../Field';
+import {AmpmField, NumberField} from './inputFields';
+
+import styles from './style.module.scss';
+
+/**
+ *
+ * @unreleased
+ */
+export default function TimePickerField() {
+    const timeObject = useWatch({name: 'createdAt'});
+
+    const {setValue} = useFormContext();
+
+    const [showFields, setShowFields] = useState<boolean>(false);
+
+    const setNewFormFieldValue = (newDateObject) => {
+        setShowFields(false);
+
+        if (newDateObject) {
+            setValue('createdAt', newDateObject, {shouldDirty: true});
+        }
+    };
+
+    return (
+        <Field label={__('Donation time', 'give')} editable onEdit={() => setShowFields(!showFields)}>
+            {showFields ? (
+                <div className={styles.timePickerPosition}>
+                    <TimeFields initialTime={timeObject} isOpen={showFields} closeFields={setNewFormFieldValue} />
+                </div>
+            ) : (
+                <span>{format(timeObject, 'h:mm a')}</span>
+            )}
+        </Field>
+    );
+}
+
+/**
+ *
+ * @unreleased
+ */
+type TimeFieldProps = {
+    isOpen: boolean;
+    closeFields: (newDate: Date | null) => void;
+    initialTime: Date;
+};
+
+export function TimeFields({isOpen, closeFields, initialTime}: TimeFieldProps) {
+    const [hours, setHours] = useState<number>(Number(format(initialTime, initialTime.getHours() >= 12 ? 'h' : 'h')));
+    const [minutes, setMinutes] = useState<number>(Number(format(initialTime, String(initialTime.getMinutes()))));
+    const [ampm, setAmpm] = useState<string>(format(initialTime, 'h:mm a').split(' ')[1]);
```

Comment:
Fun fact, the am/pm unit is called the meridiem. 🤓

Reply:
Fancy..  I like it! Let's update this.  https://github.com/impress-org/givewp/pull/6752/commits/3bf123272f798e5c17af62674640d3186137bb0e